src/d3/hapticplot.d3.spec.ts

```diff
@@ -177,3 +220,46 @@ function hasSoundTriggers(scene: HTMLElement, shape: string): boolean{
 function hasGuidance(controller: HTMLElement): boolean{
   return (controller as Entity).components.hasOwnProperty('guide');
 }
+
+// Initilizes the guide component, and returns the currInterval
+function getGuidanceInterval(controller: HTMLElement): number{
+  const controllerEntity = controller as Entity;
+  controllerEntity.components.guide .init();
+  return (controllerEntity.components.guide as Guidance).state.currInterval;
+}
+
+// Returns haptic intensity when the controller is near a data point
+function getGuidanceHapticIntensity(controller: HTMLElement): number{
+  const controllerEntity = controller as Entity;
+  controllerEntity.object3D.position.set(0, 1.5, -0.35);
+  controllerEntity.components.guide.init();
+  controllerEntity.components.guide.tick!(1, 1);
+  return (controllerEntity.components.guide as Guidance).state.intensity;
+}
+
+// Returns haptic intensity during the first half of the vibration interval when the controller is in contact with a data point
+function getIntervalIntensityFront(controller: HTMLElement): number{
+  const controllerEntity = controller as Entity;
+  controllerEntity.object3D.position.set(0, 1.7, -0.35);
+  controllerEntity.components.guide.init();
+  controllerEntity.components.guide.tick!(1, 1);
+  return (controllerEntity.components.guide as Guidance).state.intensity;
+}
+
+// Returns haptic intensity during the second half of the vibration interval when the controller is in contact with a data point
+function getIntervalIntensityBack(controller: HTMLElement): number{
+  const controllerEntity = controller as Entity;
+  controllerEntity.object3D.position.set(0, 1.7, -0.35);
+  controllerEntity.components.guide.init();
+  controllerEntity.components.guide.tick!(65, 65);
+  return (controllerEntity.components.guide as Guidance).state.intensity;
+}
+
+// Returns the current inteval value after a tick of duration 100ms
+function getIntervalReset(controller: HTMLElement): number{
+  const controllerEntity = controller as Entity;
+  controllerEntity.object3D.position.set(0, 1.7, -0.35);
+  controllerEntity.components.guide.init();
+  controllerEntity.components.guide.tick!(100, 100);
+  return (controllerEntity.components.guide as Guidance).state.currInterval;
+}
```

Comment:
In a future PR, you may want to consider moving all these utility functions into their own file.